src/pages/GeneralAdminView/GeneralAdminUserChange/GeneralAdminUserChange.jsx

```diff
@@ -1,77 +1,123 @@
-import React, { useState, useEffect } from 'react';
-import { useSelector } from 'react-redux';
-import languageSelector from '../../../assets/languages/languageSelector';
-import { toast } from 'react-toastify';
-import { AiFillDelete } from 'react-icons/ai';
-import axios from 'axios';
-import Cookies from 'universal-cookie';
-import './GeneralAdminUserChange.css'
-import { API_URL } from '../../../constants/apiUrl';
+import React, { useState, useEffect } from "react";
+import { useSelector } from "react-redux";
+import languageSelector from "../../../assets/languages/languageSelector";
+import { toast } from "react-toastify";
+import { AiFillDelete } from "react-icons/ai";
+import axios from "axios";
+import Cookies from "universal-cookie";
+import "./GeneralAdminUserChange.css";
+import { API_URL } from "../../../constants/apiUrl";
 
 const GeneralAdminUserChange = () => {
-  const language = useSelector(state => state.languageReducer);
-  const [email, setEmail] = useState('');
+  const language = useSelector((state) => state.languageReducer);
+  const [email, setEmail] = useState("");
   const [isExpanded, setIsExpanded] = useState(false);
   const [users, setUsers] = useState([]);
   const cookie = new Cookies();
-  const token = cookie.get('token');
+  const token = cookie.get("token");
   const config = {
     headers: {
-      'Authorization': `Bearer ${token}`
-    }
+      Authorization: `Bearer ${token}`,
+    },
   };
+  const [filter, setFilter] = useState("");
+  const [renderList, setRenderList] = useState([]);
 
+  useEffect(() => {
+    setRenderList([...users]);
+  }, []);
 
   const fetchUsers = async () => {
     try {
       const response = await axios.get(`${API_URL}/users`, {
-        params: { role: 'user' },
+        params: { role: "user" },
         ...config,
       });
       setUsers(response.data.data);
     } catch (error) {
-      toast.error(error.response?.data?.message || languageSelector(language, 'usersFetchError'));
+      toast.error(
+        error.response?.data?.message ||
+          languageSelector(language, "usersFetchError")
+      );
     }
   };
-  
+
   useEffect(() => {
     fetchUsers();
   }, []);
-  
+
   const toggleExpanded = () => {
     setIsExpanded(!isExpanded);
   };
 
   const handleSubmit = async (e) => {
     e.preventDefault();
     try {
-      const response = await axios.put(`${API_URL}/users/change-role`, {
-        email,
-        user_role: 'appAdmin',
-      }, config);
-      setEmail('');
+      const response = await axios.put(
+        `${API_URL}/users/change-role`,
+        {
+          email,
+          user_role: "appAdmin",
```

Comment:
this user_role value could be better in an imported constant and implemented here, for security.